modules/hello-world/examples/KvBulkHelloWorld.java

```diff
@@ -0,0 +1,152 @@
+      List<MutationResult> results = Flux.fromIterable(documents)
+          .flatMap(document -> reactiveCollection.insert(
+              document.getId(), document.getContent()
+            )
+          )
+          .collectList()
+          .block(); // Wait until all operations have completed.
+
+      // Print all the results.
+      for (MutationResult result : results) {
+        System.out.println("CAS: " + result.cas());
+      }
+      // end::kv-bulk-insert[]
+    }
+
+    {
+      System.out.println("\nExample: [kv-bulk-upsert]");
+      // tag::kv-bulk-upsert[]
+      JsonObject newUser1 = JsonObject.create().put("id", "user_111").put("email", "tom@gmail.com");
+      JsonObject newUser2 = JsonObject.create().put("id", "user_222").put("email", "jerry@gmail.com");
+      JsonObject newUser3 = JsonObject.create().put("id", "user_333").put("email", "mickey@gmail.com");
+      
+      List<JsonDocument> newDocuments = new ArrayList<>();
+      newDocuments.add(new JsonDocument("user_111", newUser1));
+      newDocuments.add(new JsonDocument("user_222", newUser2));
+      newDocuments.add(new JsonDocument("user_333", newUser3));
+
+      // Iterate over a list of documents to upsert.
+      List<MutationResult> results = Flux.fromIterable(newDocuments)
+          .flatMap(newDocument -> reactiveCollection.upsert(
+              newDocument.getId(), newDocument.getContent()
+            )
+          )
+          .collectList()
+          .block(); // Wait until all operations have completed.
+
+      // Print all the results.
+      for (MutationResult result : results) {
+        System.out.println("CAS: " + result.cas());
+      }
+      // end::kv-bulk-upsert[]
+    }
+
+    {
+      System.out.println("\nExample: [kv-bulk-get]");
+      // tag::kv-bulk-get[]
+      // Iterate over a list of documents to fetch.
+      List<GetResult> results = Flux.fromIterable(documents)
+          .flatMap(document -> reactiveCollection.get(document.getId()))
+          .collectList()
+          .block(); // Wait until all operations have completed.
+
+      // Print all the results.
+      for (GetResult result : results) {
+        JsonObject document = result.contentAsObject();
+        System.out.println("Document: "  + document);
+        System.out.println("CAS: " + result.cas());
+      }
+      // end::kv-bulk-get[]
+    }
+
+    {
+      System.out.println("\nExample: [kv-bulk-remove]");
+      // tag::kv-bulk-remove[]
+      // Iterate over a list of documents to remove.
+      List<MutationResult> results = Flux.fromIterable(documents)
+          .flatMap(document -> reactiveCollection.remove(document.getId()))
+          .collectList()
+          .block(); // Wait until all operations have completed.
+
+      // Print all the results.
+      for (MutationResult result : results) {
+        System.out.println("CAS: " + result.cas());
+      }
+      // end::kv-bulk-remove[]
+    }
+  }
+}
+
+
+// tag::kv-bulk-class[]
+class JsonDocument {
```

Comment:
Is it best practice to define a class to store the tuple of id/value?

(e.g. as opposed to just using a Map and iterating over key/values?
Hmm, it's Java, so I guess an explicit new class may well be idiomatic... just seemed a bit overkill)

Reply:
💡 trying to implement similar thing and have seen the light...

Reply:
Yes, in Java you'd usually create a class for this.  But Java 14 added Records, which can do this with minimal boilerplate, so I'd use those now.

Reply:
Worth discussing in our docs sync meeting with PMs, for now will leave as is, but would be really nice to update this to use Records!